app/controllers/api/v1/rooms_controller.rb

```diff
@@ -6,11 +6,12 @@ class RoomsController < ApiController
       skip_before_action :ensure_authenticated, only: %i[public_show]
 
       before_action :find_room, only: %i[show update destroy recordings recordings_processing purge_presentation public_show]
-
+      before_action except: %i[public_show] do
```

Comment:
I added the `except` after looking at the rspec test.
I don't fully understand why the `session[:user_id] = nil` in the rspec test for this. Could someone explain the usage of `public_show`

Reply:
`session[:user_id] = nil` is to log off the current_user. So in specs it is usually to make sure some one not logged in does not have the access to something.

Reply:
Yeah i'm aware that it signs out the user. I'm more looking for why do that for this specific test and why do they still have access to `public_show` even thought they are not logged in

Reply:
Right. I'm also unsure of what is going on with `public_show` (and `publicRoom` from #3836)